File: Ruby/two_sum.rb
# def two_sum(nums, target)
#   for first_num in (0..nums.length-1)
#       for second_num in (first_num+1..nums.length-1)
#           if nums[second_num] == target - nums[first_num]
#               return [first_num, second_num]
#           end
#       end
#   end
# end

def two_sum(nums, target)
  hashmap = {}
    for i in (0..nums.length-1)
      current = target - nums[i]
      if hashmap.include?(current)
        return [i, hashmap[current]]
      end
      hashmap[nums[i]] = i
    end
end


p two_sum([2, 7, 11, 15], 9)
p two_sum([3,2,4], 6)
p two_sum([3, 3], 6)

File: Python/two_sum.py
# ex1
# current = target - x
# nums[j] == target - nums[i]
# def two_sum(nums, target):
#   for i in range(len(nums)):
#     for j in range(i+1, len(nums)):
#       if nums[j] == target - nums[i]:
#         return [i, j]   

def two_sum(nums, target):
  hashmap = {}
  
  for i in range(len(nums)):
    current = target - nums[i]
    if current in hashmap:
      return [i, hashmap[current]]
    hashmap[nums[i]] = i
  
print(two_sum([2, 7, 11, 15], 9))
print(two_sum([3,2,4], 6))
print(two_sum([3, 3], 6))